curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -2243,17 +2250,23 @@ CURVEFS_ERROR DataCache::Flush(uint64_t inodeId, bool toS3) {
                 }
                 VLOG(9) << "PutObjectAsyncCallBack: " << context->key
                         << " pendingReq is: " << pendingReq;
+                if (useReadCacheOnly) {
+                    VLOG(9) << "Write to read cache, name: " << context->key;
+                    s3ClientAdaptor_->GetDiskCacheManager()
+                                    ->Enqueue(context, true);
+                }
                 return;
             }
             LOG(WARNING) << "Put object failed, key: " << context->key;
             s3ClientAdaptor_->GetS3Client()->UploadAsync(context);
         };
 
     std::vector<std::shared_ptr<PutObjectAsyncContext>> uploadTasks;
-    bool useDiskCache =
+    bool useReadWriteCache =
```

Comment:
`useReadCacheOnly` 和 `useReadWriteCache`
Is it possible to add an enum, like this:
```c++
enum CachePolicy {
    NCache,
    RCache,
    WRCache,
}
```
To express caching strategy?

Reply:
> A good idea.
> 
> So how about now？

In fact, the best way is bit operation, that is, 0 means NCache, 1 means RCache, 2 means WCache, and 3 means WRcache. Bitwise AND to calculate whether to open the corresponding operation.
```
enum CachePolicy {
    NCache = 0,
    RCache = 1<<0,
    WCache = 1<<1,
    WRCache=RCache&WCache,
};
```
Whether to enable read cache:
```
if (cachePoily & CachePolicy::RCache) {
    ....
}
```
But this will lose the readability of the code.